src/dymaptic.GeoBlazor.Core/Scripts/authentication.ts

```diff
@@ -0,0 +1,55 @@
+// authentication imports
+import IdentityManager from "@arcgis/core/identity/IdentityManager";
+import OAuthInfo from "@arcgis/core/identity/OAuthInfo";
+import {arcGisObjectRefs} from "./arcGisJsInterop";
+
+export default class OAuthAuthenticationWrapper {
+    private appId: string;
+    private info: OAuthInfo;
+
+    constructor(dotNetReference, apiKey) {
```

Comment:
I'm also confused here. Why pass in anything we're not using, if we've guaranteed that `initialize` is always called before other methods?

Reply:
This is setup to match the call used by `LogicComponent` which, itself might need to be refactored. (Currently, ApiKey is passed through to classes like `projection.ts` where they are attached to the `esriConfig` object. I assume this is to deal with a similar problem that I'm having, and that's ensuring that the authentication is set before you do anything). There is a pretty strong dependency on ApiKey there. But when logged in via OAuth, there is no ApiKey, there is only the token, and that expires regularly so you can't pass it around. Instead Esri just "takes care of this" under the hood, attaching the correct token before the call is sent out.

There's actually a complexity here we haven't even discussed, and that's connecting to a server that is federated with a portal. In that scenario, you would first have to go to portal and ask for a token specifically for that server, and then use that for any requests for that server (until the token expires). Fortunately for us, the JS API takes care of all of this for us, we just have to let it.

I had to accept dotNetReference and apiKey in the constructor b/c that's what `InvokeAsync` and `InvokeVoidAsync` both do in `LogicComponent` and I wanted to follow the same pattern. I'm dropping the dotNetReference b/c I don't need to call back, but it's there for the future if we do ever want to do that (like to raise an exception).